### ISSUE TITLE
Split CI tests

### ISSUE THREAD

**Issue by jaimergp** [openforcefield/openff-toolkit#561]
# Is your feature request related to a problem? Please describe.

Current CI logs for the actual test is a single category with a loooong list of items addressing different aspects of the codebase and its quality: unit tests, documentation, examples, notebooks...

# Describe the solution you'd like

GH Actions provide UI elements to easily distinguish categories and concerns. We can have multiple _workflows_ (separate YML files), _jobs_ (separate elements in the PR UI), and _steps_ (collapsible blocks within each job). Each of these elements gets a failed/success status, so we could easily see what part is failing. We can benefit from that visual aid if we split our current monolithic "test everything" step. My proposal (hierarchy is workflow > step)

* **Testing**. This workflow will have several _jobs_, one for each entry in the build matrix. Each _job_ would have these testing _steps_.
    * Unit tests: guarantee the library is working as expected (this is the main category, and the only one that runs across all entries in the build matrix)
    * Run examples and notebooks
* **Documentation**. Single job per workflow, so it runs only once per push. Steps would include:
    * Ensure sphinx builds (no deploy, just run the build part to make sure we don't break it silently)
    * Ensure all links are alive (might fail for network errors and this is not critical)
* **Style**. Single job per workflow, so it runs only once per push. Steps would include:
    * Run some linters (already covered by LGTM?)
    * Run formatting checks

# UI examples

Different workflows / jobs:

![image](https://user-images.githubusercontent.com/2559438/78231445-18ba3780-74d3-11ea-9938-6e6c37981da7.png)

Different steps within a job:

![image](https://user-images.githubusercontent.com/2559438/78231473-253e9000-74d3-11ea-8301-e51d7a1fb5b7.png)

**Comment by jaimergp** [comment]
I'll use the upcoming PR to add the README badges too, which I forgot in #554 😬

**Comment by mattwthompson** [comment]
Looks like we mostly pulled this off:

* **Testing**: Tests are split up between unit tests, example notebooks, checking README links, and running code snippets in docs: https://github.com/openforcefield/openforcefield/blob/77b17c71d58f5e95808d0d984a42bf812a2b4040/.github/workflows/CI.yml
* **Documentation**: Not automated here, as far as I can tell.
* **Linting**: Minimal linting action (using only `pip`!): https://github.com/openforcefield/openforcefield/blob/77b17c71d58f5e95808d0d984a42bf812a2b4040/.github/workflows/lint.yml

**Comment by mattwthompson** [comment]
I think this was done a while back (thanks Jaime!)
